esp/services/WsDeploy/WsDeployService.cpp

```diff
@@ -4665,7 +4665,13 @@ bool CWsDeployFileInfo::handleEspServiceBindings(IEspContext &context, IEspHandl
         if (resource)
           xpath.appendf("[@resource='%s']", resource);
 
-        IPropertyTree* pSubType = pEnvRoot->queryPropTree(xpath.str());
+        IPropertyTree* pSubType = nullptr;
+        Owned<IPropertyTreeIterator> iterSubType = pEnvRoot->getElements(xpath.str());
```

Comment:
The reason for using an iterator, even though I only want the first element, is because queryPropTree() returns an ‘ambigous xpath exception” when provided a valid xpath using an index.  I do a hasProp call to verify that the element is there.  The queryProp() call then throws an exception even though the index is provided and the xpath is valid and not ambigous.  This is the output from the debugger:
```
(gdb) p pEnvRoot->hasProp("./Software/EspProcess[@name='myesp']/EspBinding[@defaultForPort='true'][@name='EspBinding'][@port='8010'][@protocol='http'][@resourcesBasedn='ou=SMC,ou=EspServices,ou=ecl'][@service='EclWatch'][@workunitsBasedn='ou=workunits,ou=ecl']/Authenticate[@resource=''][1]")
$16 = true
(gdb) p pEnvRoot->queryProp("./Software/EspProcess[@name='myesp']/EspBinding[@defaultForPort='true'][@name='EspBinding'][@port='8010'][@protocol='http'][@resourcesBasedn='ou=SMC,ou=EspServices,ou=ecl'][@service='EclWatch'][@workunitsBasedn='ou=workunits,ou=ecl']/Authenticate[@resource=''][1]")
The program being debugged entered a std::terminate call, most likely caused by an unhandled C++ exception.  GDB blocked this call in order to prevent the program from being terminated, and has restored the context to its original state before the call.
To change this behaviour use "set unwind-on-terminating-exception off".
Evaluation of the expression containing the function (PTree::queryProp(char const*) const)
will be abandoned.
```


I can replace the ForEach macro,  but I think the iterator needs to stay, and some of the ugliness.